<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>907</width>
    <height>586</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Ray Marching Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/res.qrc">
    <normaloff>:/res/img/vls.png</normaloff>:/res/img/vls.png</iconset>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="RenderWidget" name="renderWidget" native="true">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>640</width>
          <height>360</height>
         </size>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout"/>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>24</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cancelButton">
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>907</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSaveAllShaders"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuExport">
    <property name="title">
     <string>Export</string>
    </property>
    <addaction name="actionTake_screenshot"/>
    <addaction name="actionExport_as_video"/>
    <addaction name="separator"/>
    <addaction name="actionExport_as_Linux_Demo"/>
   </widget>
   <widget class="QMenu" name="menuCamera">
    <property name="title">
     <string>Camera</string>
    </property>
    <addaction name="actionInsert_camera_keyframe"/>
    <addaction name="actionReset_camera"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuExport"/>
   <addaction name="menuCamera"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/new.png</normaloff>:/res/img/new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/open.png</normaloff>:/res/img/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveAllShaders">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/save.png</normaloff>:/res/img/save.png</iconset>
   </property>
   <property name="text">
    <string>Save shaders</string>
   </property>
   <property name="toolTip">
    <string>Save all shaders</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionReset_camera">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/camera.png</normaloff>:/res/img/camera.png</iconset>
   </property>
   <property name="text">
    <string>Reset camera</string>
   </property>
  </action>
  <action name="actionTake_screenshot">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/screenshot.png</normaloff>:/res/img/screenshot.png</iconset>
   </property>
   <property name="text">
    <string>Take screenshot</string>
   </property>
  </action>
  <action name="actionInsert_camera_keyframe">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/record_camera_keyframe.png</normaloff>:/res/img/record_camera_keyframe.png</iconset>
   </property>
   <property name="text">
    <string>Insert camera keyframe</string>
   </property>
   <property name="toolTip">
    <string>Insert/set current camera parameters into keyframe</string>
   </property>
   <property name="shortcut">
    <string>I</string>
   </property>
  </action>
  <action name="actionExport_as_video">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/export_video.png</normaloff>:/res/img/export_video.png</iconset>
   </property>
   <property name="text">
    <string>Export as Video</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Export as Video&lt;span style=&quot; font-style:italic;&quot;&gt; (render.mp4)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="actionExport_as_Linux_Demo">
   <property name="icon">
    <iconset resource="../resources/res.qrc">
     <normaloff>:/res/img/export_linux.ico</normaloff>:/res/img/export_linux.ico</iconset>
   </property>
   <property name="text">
    <string>Export as Linux Demo</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RenderWidget</class>
   <extends>QWidget</extends>
   <header>renderwidget.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/res.qrc"/>
 </resources>
 <connections/>
</ui>
